### ISSUE TITLE
Release 1.15.0 changed behavior of --skip=""

### ISSUE THREAD

**Issue by CecileRobertMichon** [onsi/ginkgo#780]
Before release 1.15, `--skip=""` would result in no tests being skipped (and only tests that match the `--focus` flag to run). Now, `--skip=""` causes all tests to be skipped.

I suspect it might be caused by this recent change: #736 

Before:
https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_cluster-api-provider-azure/1216/pull-cluster-api-provider-azure-e2e-full/1371635147035643904
```
/home/prow/go/src/sigs.k8s.io/cluster-api-provider-azure/hack/tools/bin/ginkgo-v1.14.2 -v -trace -tags=e2e -focus="Workload cluster creation" -skip="" -nodes=3 --noColor=false  ./test/e2e -- \
    	-e2e.artifacts-folder="/logs/artifacts" \
    	-e2e.config="/home/prow/go/src/sigs.k8s.io/cluster-api-provider-azure/test/e2e/config/azure-dev-envsubst.yaml" \
    	-e2e.skip-resource-cleanup=false -e2e.use-existing-cluster=false 

Ran 9 of 19 Specs in 5065.109 seconds
SUCCESS! -- 9 Passed | 0 Failed | 0 Pending | 10 Skipped
```    

After:
https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/directory/pull-cluster-api-provider-azure-e2e-full/1371598441024589824
```
/home/prow/go/src/sigs.k8s.io/cluster-api-provider-azure/hack/tools/bin/ginkgo-v1.15.1 -v -trace -tags=e2e -focus="Workload cluster creation" -skip="" -nodes=3 --noColor=false  ./test/e2e -- \
    	-e2e.artifacts-folder="/logs/artifacts" \
    	-e2e.config="/home/prow/go/src/sigs.k8s.io/cluster-api-provider-azure/test/e2e/config/azure-dev-envsubst.yaml" \
    	-e2e.skip-resource-cleanup=false -e2e.use-existing-cluster=false 
    	
Ran 0 of 19 Specs in 144.087 seconds
SUCCESS! -- 0 Passed | 0 Failed | 0 Pending | 19 Skipped
```

from testing https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/1216

**Comment by onsi** [comment]
thanks for reporting this @CecileRobertMichon - i'll push out a fix and cut a new patch version today.

**Comment by onsi** [comment]
fixed in v1.15.2
